lib/plugins/aws/deploy-function.js

```diff
@@ -267,15 +267,46 @@ class AwsDeployFunction {
     if (functionObj.layers && !functionObj.layers.some(_.isObject)) {
       params.Layers = functionObj.layers;
     }
-    if (
-      params.Layers &&
-      remoteFunctionConfiguration.Layers &&
-      _.isEqual(
-        new Set(params.Layers),
-        new Set(remoteFunctionConfiguration.Layers.map((layer) => layer.Arn))
-      )
-    ) {
-      delete params.Layers;
+
+    // Check if we have remotely managed layers and add them to the update call
+    // if they exist in the remote function configuration.
+    const isConsoleSdkLayerArn = RegExp.prototype.test.bind(
+      /(?:177335420605|321667558080):layer:sls-/u
+    );
+    const serverlessConsoleLayerArns = (remoteFunctionConfiguration.Layers || [])
+      .filter(({ Arn: arn }) => isConsoleSdkLayerArn(arn))
+      .map(({ Arn }) => Arn);
+    const hasServerlessConsoleLayers = serverlessConsoleLayerArns.length > 0;
+    if (!functionObj.layers || (functionObj.layers && !functionObj.layers.some(_.isObject))) {
```

Comment:
To avoid double checking for object referernces, I would rather update the check at L267, so it  unconditionally creates `params.Layers` in above case.  Then here we can just check `if (params.Layers)`

Reply:
This is different from the check on `L267` though. `L267` is checking that `functionObj.layers` exists and does not have any objects.

This if statement is saying we should run this block if `functionObject.layers` does not exist or if it exists with no objects.

It is also important that it is wrapping this block because if there are objects and the function is console managed we don't want to execute the code below.

I probably need to add a unit test for this to highlight this intended functionality.

Reply:
@medikoo I added a unit test for this scenario 👇

https://github.com/Danwakeem/serverless/blob/ccff42ea8b5d3a500c52f47b6f0d569fb96a555d/test/unit/lib/plugins/aws/deploy-function.test.js#L891-L895

Reply:
Okay I think I figured out a way to combine the two into a single if statement 👍